MyString.java

```diff
@@ -36,8 +48,32 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
-        return false;
+        for (int i=0;i<str1.length();i++)
+        {
+            char c=str1.charAt(i);
+            int str1_c=counter_c(str1, c);
+            int str2_c=counter_c(str2, c);
+            if(str1_c>str2_c)
+            {
+                return false;
+            }
+            if (str2.indexOf(c)==-1) {
+                return false;
+            }
+        }
+        return true;

```

Comment:
nice, could be improved, by checking if this is first appearance of letter in str1.
`if (i == str1.indexOf(str.charAt(i)))`